aea/connections/base.py

```diff
@@ -100,22 +101,25 @@ def __init__(
         )
 
     @property
-    def loop(self) -> Optional[AbstractEventLoop]:
+    def loop(self) -> asyncio.AbstractEventLoop:
         """Get the event loop."""
-        return self._loop
-
-    @loop.setter
-    def loop(self, loop: AbstractEventLoop) -> None:
-        """
-        Set the event loop.
-
-        :param loop: the event loop.
-        :return: None
-        """
-        assert (
-            self._loop is None or not self._loop.is_running()
-        ), "Cannot set the loop while it is running."
-        self._loop = loop
+        assert asyncio.get_event_loop().is_running()
```

Comment:
can I ask you why the loop must be running? In some implementations, I might just want to get the loop.
The check could be done either manually or by having another property like `ensure_running_loop` that does the check.

Reply:
... maybe `loop` and `running_loop`, respectively?

Or even, don't have loop properties at all. Connection developers can just use `asyncio.get_event_loop` and `asyncio.get_running_loop` respectively. Do you see benefits, or is that too extreme?

Reply:
probably this check is redundant and kind of legacy from the previous implementation

Reply:
also this should help to avoid loop usage outside of asyncio coroutines.

so 
```
class Connection:
   def __init__():
        self.arg = something(self.loop)
```

will fail if __init__ called outside of asyncio coroutine